### DISCUSSION TITLE
Multiple databases support

### DISCUSSION THREAD

**Discussion by deividaspetraitis** [benbjohnson/wtf#62] [Q&A] [answered]
Hello Ben,

Thanks for sharing these great resources!
I was wondering how you would handle a case adding new database support say let's say leaving User fetching as it is in `sqllite` and [UserAuths](https://github.com/benbjohnson/wtf/blob/bcc6c727f98398d48fd8bfbf6e73343b18ff68c1/user.go#L26) from another database or even external service API.

Would it be good approach inject DB/API service handler into [UserService](https://github.com/benbjohnson/wtf/blob/bcc6c727f98398d48fd8bfbf6e73343b18ff68c1/sqlite/user.go#L19) ?

Thank you,

Deividas

**Comment by benbjohnson** [comment, marked as answer]
Hi @deividaspetraitis. If the `sqlite.UserService` needed to interact with an `AuthService` then I'd probably attach it as a field on `sqlite.UserService`:

```go
package sqlite

type UserService struct {
	AuthService myapp.AuthService
}
```

Then I'd attach the concrete implementation in the `cmd/myapp/main.go` so that the sqlite implementation wouldn't need to know anything about the other database implementation.

**Reply by deividaspetraitis** [reply]
Thank you @benbjohnson for sharing your thoughts!
